### DISCUSSION TITLE
Power calibration doesn't work (it's way too far off)

### DISCUSSION THREAD

**Discussion by Pareidol** [arendst/Tasmota#14636] [Q&A] [answered]
I have this plug with power monitoring: https://templates.blakadder.com/kogeek_W-DEXI.html
After  installing the latest official tasmota release I installed the template. The relay, button and led seems to work.
Then I tried to do the calibration process ( https://tasmota.github.io/docs/Power-Monitoring-Calibration ).
I measured my Voltage (=227V) entered it in the console and tasmota show me the 227V in the live values (maybe some Volts up or down). When I enter the power measured by my Kill-A-Watt (=58 Watt) Tasmota shows me only 9 Wall as live value. I assume that is way too far off to use the fine tuning. What could be wrong?

**Comment by Jason2866** [comment]
What load have you connected for calibrating. Only a resistive load works. It should be at least 60W.
Better would be a toaster.

**Comment by Pareidol** [comment]
I have a 60 Watt lightbulb like described in the calibration process.
But I found another strange behaviour, and maybe this is the reason why it is so hard to calibrate.

- I've the same plug with original firmware: It measures acceptably accurate and if I unplug the load and switch on the relais, it measures nothing, as expected. 
- If I use the same plug with Tasmota installed, it measures power as soon as the relais is switched on even if there is no load connected to the plug. 

Cloud it be an error in the template? Seems as it measures the current draw of the relais or so.

**Reply by barbudor** [reply]
Depending if the power measurment is located after or before the internal power supply for the smart part, the measurment may include the plug's own consumption.
However the power used by the plug itself should be neglictable vs the power used by the load

For the calibration you told us about calibrating the voltage. But did you calibrate the power/current also ?
The higher is the power of the calibration load, the more accurate calibration will be.

**Comment by Pareidol** [comment]
I flashed Tasmota again to have a "factory reset" and set up the device completely new. Then I did  the complete calibration process with the help of a Kill-A-Watt. Adjusted Voltage, Current, Power. After a little fiddeling around I got nearly the same readings as the Kill-A-Watt. But then I recognized the strange measuring behaviour.

If I activate the load relais, I get the reading e.g. 60 Watts, and if I unplug the load (relay stays activated) I have exactly the same reading. And I would assume, it doesn't matter where the measure point is, the power consumption of the device itself should be nearly nothing in comparison to the load.

I soldered a new LED to the board on GPIO2 (need it as an additional signal light), added it as second relais, and even there, If I activate this little LED, It shows the exact same power consumption.

![2022-01-28_20h25_56](https://user-images.githubusercontent.com/94805419/151615199-13d00364-61ac-4be4-8e47-0e92c1c2367f.png)



This is the Template I use (based on a template based on Blackadder)
{"NAME":"W-DEXI","GPIO":[0,64,1,0,2720,2656,0,0,2592,288,224,0,0,0],"FLAG":0,"BASE":18}

**Reply by barbudor** [reply]
Very strange
I have Gosund SP111/NousA1 plugs using the same measurement chip, although a different template, and it works as expected.
The fact that the power remains there even when removing the load is strange. I don't have the same behavior on my plugs.

It could be possible that the design of the board has changed and template is invalid.
Are those plugs easy to open ? So you could try to check on the board the connections ?

**Reply by Pareidol** [reply]
Not really easy to open, heavily glued. But I ha one cracked open to flash it serially (Tuya-convert didn't work for me). But sadly glued it already back together. Maybe I'll open it again :-(

**Comment by Jason2866** [comment, marked as answer]
The Gpio config has probably changed (new circuit design). Voltage measurement can work but current is wrong.

**Reply by barbudor** [reply]
This plug is similar in form and name but is using `HLWBL SELi` instead of `HLWBL SEL`
May be worth to try
https://templates.blakadder.com/minleaf_W-DEXI.html

do not change the buttton if it works correctly, just GPIO12

**Reply by Pareidol** [reply]
Tried this, no success so far, maybe I need to open the plug again :-(

**Reply by barbudor** [reply]
If you had a hard time to crackit open, you may also have been unlucky and damaged something. Would be a pity.
But let's see if you can check the internal wiring.

**Reply by Pareidol** [reply]
Defenitely no damage on the hardware (just the casing was hard to open because it was glued). Tried a second one, same behaviour. 
The current sensor is a HLW8012, and not the one from the template. But even with that setting, no luck so far.

Measured the PCB lines, they seem to fit to the Pins in the template, but always the same, as soon as the internal relays is enabled, it measures current, even if there is no load connected. I invested now mor than 10 hours trying to solve that issue. Seems as I need to look for another sollution, it's not worth it. Can't say what causes the issue. From my opinion, it's caused by Tasmota, because the original software works fine.

**Reply by barbudor** [reply]
If you start to see current as soon as Power is ON, even with no load connect, that looks like a lot V and I measurments are swapped, means either the GPIO assignements are not correct or not using HLWBL_SEL / HLWBL SEL*i* correctly
These are pins of the HLW8012:
![image](https://user-images.githubusercontent.com/2996042/151703264-bbe12c73-ae46-4735-8ad4-a3be9b72fb66.png)
![image](https://user-images.githubusercontent.com/2996042/151703285-34f2358c-6cf6-4933-8934-2f015c294769.png)

Can't help much more without having the plug in hands

**Reply by Pareidol** [reply]
Tried every assignment I could imagine.... no luck so far. If you would live in Germany I would send you one, because it drives me nuts, that I invested so much time with no result in the end :-/
